core/src/ids.rs

```diff
@@ -210,6 +210,34 @@ impl ProgramId {
         let argument = [SALT, code_id.as_ref(), salt].concat();
         hash(&argument).into()
     }
+
+    /// Generate ProgramId from given CodeId, MessageId and salt
+    pub fn generate_for_pallet(code_id: CodeId, salt: &[u8], message_id: MessageId) -> Self {
```

Comment:
we don't need that fn at all.
just rename already existing `generate` into `generate_from_user` and use it only on pallet